### ISSUE TITLE
L2 in "Soft Localization learning loss"

### ISSUE THREAD

**Issue by Lilyo** [SchroeterJulien/2020-ICML-UDL-Workshop-Robust-Temporal-Point-Event-Localization#1]
Thank you for your work firstly, I am curious about why you choose L2 as criteria in "Soft Localization learning loss" instead of L1?
What good properties of L2 do you consider?

**Comment by SchroeterJulien** [comment]
Thank you for your interest in the paper. 

Loss functions based on (2D) heatmap comparison are extremely common in computer vision (e.g., human pose estimation, facial landmark detection). While many alternatives have been proposed, the L2-norm appears to be the default choice in the field [1,2,3]. Thus, as this component of our loss is a 1D variant, the L2-norm seems a rather natural choice. In terms of properties, the L2-norms penalizes proportionally more deviations in regions with higher event densities than in regions where only a single event occurs. While putting more emphasis on more complex (temporal) regions seems a sound choice, determining whether this feature has a negative or positive impact on the training and more importantly on robustness still requires further investigation. Another aspect of the L2-norm is that it is continuously differentiable and that it allows for much easier manual computation of the gradient. While this can seem minor because of _autograd_, several applications require manual gradient computation. In terms of practical applications, if the main concern is to tailor the model as much as possible to the task at hand, the choice of the norm can easily be integrated into the parameter search. 

Hope this helps.

[1] Newell, A., Yang, K. and Deng, J., 2016. Stacked hourglass networks for human pose estimation. In ECCV. 
[2] Tompson, J., Goroshin, R., Jain, A., LeCun, Y. and Bregler, C., 2015. Efficient object localization using convolutional networks. In CVPR.
[3] Xiao, B., Wu, H. and Wei, Y., 2018. Simple baselines for human pose estimation and tracking. In ECCV.

**Comment by Lilyo** [comment]
Thank you for your clarification, it is very clear! However, I have other questions, I will open a new issue.
